### DISCUSSION TITLE
Default filename for saving a printed invoice needs customer name

### DISCUSSION THREAD

**Discussion by sbts** [ledgersmb/LedgerSMB#5847] [Ideas]
When "printing" an invoice the associated filename (LaTex, HTML, or other) currently follows the pattern
`invoice-$invoice_number`
it would be preferable to append a sanitized version of the entity name so it looks something like.....
`invoice-$invoice_number-joe_s_fish_chips`
where the original customer name was
`Joe's Fish & Chips`
Essentially converting any sequence of non alpha numeric chars to a single `_`

**Comment by nick-prater** [comment]
I think this is a good idea, but I'd prefer to look at this from my customers' point of view. For them, it's more useful to have my company name in the invoice filename, so it's obvious who the invoice is _from_. Perhaps it's possible to include both in the file name?

**Comment by ehuelsmann** [comment]
I'd like to solve the part of the request for the own company by advocating a configurable prefix. As for the customer, is like to propose to use the customer number instead of the name: it's meaningful and relatively short.

**Comment by freelock** [comment]
I was just thinking about this for an aging report -- I'd like to include the customer name in the filename for this run. Maybe make some settings in the default settings, for filename patterns? And support some tokens for customer names...

Not to get too complex, but I keep thinking of what using a template system brings to this -- not sure about Template Toolkit, but Mustache (JS), Twig (PHP), Smarty (PHP), Jinja2 (Python) all allow further processing for things like truncating a string after a certain length, forcing lowercase (or capitalizing the first word).

Maybe we can incorporate something like that?

**Reply by ehuelsmann** [reply]
Yes, template toolkit has all of those. Check http://template-toolkit.org/docs/manual/Filters.html and see "lcfirst", "truncate()", etc.
